### DISCUSSION TITLE
Weird behavior / crash after import of bulk of images

### DISCUSSION THREAD

**Discussion by lunatikk1984** [photoprism/photoprism#4655] [Q&A]
Hi everyone,

I setup photoprism in "Container Station" of my QNAP ts-251d NAS. Last week I started the import of a total of ~70K images/videos. The import completed after a couple of days. However I now saw that photoprism is in a state where it's non-functional. 
After the import it starts writing stuff on the disk. I don't see any log outputs that indicate what it's doing exactly. When I try to log-in into the web frontend it says "wrong credentials". Sometimes the login works, though.
This happens also after I restart the container and import 2 or 3 images.

For instance this is what the log says after the import was completed:
`
INFO[2024-12-05T09:13:07Z] import completed in 60 s

WARN[2024-12-05T09:24:10Z] file fsnrxkq48q943zds: database is locked while loading markers 
INFO[2024-12-05T09:25:04Z] photo: 2014/12/20141213_000000_C98A5C2E keeps existing meta title 
INFO[2024-12-05T09:40:53Z] libx264: transcoding /photoprism/storage/cache/media/9/9/f/99f9e7b6df612fd6c013872483e46db96a29571e.mp4 to avc 
INFO[2024-12-05T09:41:03Z] libx264: created 99f9e7b6df612fd6c013872483e46db96a29571e.mp4.avc [10.074525376s] 
WARN[2024-12-05T09:50:37Z] sync: database is locked                     
WARN[2024-12-05T09:50:37Z] share: database is locked  
`
so directly after the import at around 9:13 the hdds started to go off like crazy (I can here them constantly working)
Then ~40m later at 9:50 I get warnings that say database is locked.

I used photoprism/photoprism:latest image in "Container Station" and only adjusted network settings for static ip and mapped the storage, originals and import folder to corresponding folders on the nas that I could access remotely.

I don't even know where to start with troubleshooting and hope that someone can help me out.

Best regards
L

**Comment by graciousgrey** [comment]
Hello :wave: 

Database locks are a known issue with Sqlite. You can find more information here:

- https://docs.photoprism.app/getting-started/troubleshooting/sqlite
- https://docs.photoprism.app/getting-started/faq/#should-i-use-sqlite-mariadb-or-mysql

**Reply by lunatikk1984** [reply]
is sqlite or mariadb active by default?

**Reply by graciousgrey** [reply]
This is specified in your compose.yaml file.

Our example files have mariadb pre-configured:
- https://docs.photoprism.app/getting-started/docker-compose/ --> https://dl.photoprism.app/docker/compose.yaml
- https://docs.photoprism.app/getting-started/portainer/ --> https://dl.photoprism.app/docker/portainer/stack.yml

**Reply by lunatikk1984** [reply]
> This is specified in your compose.yaml file.
> 
> Our example files have mariadb pre-configured:
> 
> * https://docs.photoprism.app/getting-started/docker-compose/ --> https://dl.photoprism.app/docker/compose.yaml
> * https://docs.photoprism.app/getting-started/portainer/ --> https://dl.photoprism.app/docker/portainer/stack.yml

Hi graciousgrey & thanks for the answer!

I did'nt use a compose.yaml explicitely, I just selected the official (latest) docker container in the QNAP container management tool (Container station) that points to https://hub.docker.com/r/photoprism/photoprism and as I didn't change anything besides network and folder mappings I was under the assumption that mariadb would be active.

This is what I see when I create the photoprism container as I did before:
![image](https://github.com/user-attachments/assets/18a8afe7-b10d-4430-8b51-ba4568f88de3)

Still I'd like to know what photoprism is doing in the background. Every now and then I see a db lock in the log but the HDDs are running and working constantly (until I stop or restart the container). See the updated log file:
`
INFO[2024-12-05T09:13:06Z] import: imported 11 files     
               
INFO[2024-12-05T09:13:07Z] import completed in 60 s                     
WARN[2024-12-05T09:24:10Z] file fsnrxkq48q943zds: database is locked while loading markers 
INFO[2024-12-05T09:25:04Z] photo: 2014/12/20141213_000000_C98A5C2E keeps existing meta title 
INFO[2024-12-05T09:40:53Z] libx264: transcoding /photoprism/storage/cache/media/9/9/f/99f9e7b6df612fd6c013872483e46db96a29571e.mp4 to avc 
INFO[2024-12-05T09:41:03Z] libx264: created 99f9e7b6df612fd6c013872483e46db96a29571e.mp4.avc [10.074525376s] 
WARN[2024-12-05T09:50:37Z] sync: database is locked                     
WARN[2024-12-05T09:50:37Z] share: database is locked                    
WARN[2024-12-05T10:05:37Z] share: database is locked                    
WARN[2024-12-05T10:05:37Z] sync: database is locked                     
INFO[2024-12-05T10:28:20Z] cell: added place hr:JFgtXRrhrxob [197.646536ms] 
WARN[2024-12-05T10:50:38Z] share: database is locked                    
INFO[2024-12-05T11:23:27Z] cell: added place de:dAZoOABHGvsp [186.667579ms] 
WARN[2024-12-05T11:35:37Z] sync: database is locked                     
WARN[2024-12-05T11:35:37Z] share: database is locked 
`

**Reply by graciousgrey** [reply]
The screenshot looks like you have configured sqlite instead of mariadb. If you are not using our example files you need to do the configuration manually.

>  Still I'd like to know what photoprism is doing in the background.

I would assume that after importing your files get indexed.

This troubleshooting guide might also be helpful to solve your locking issues: https://docs.photoprism.app/getting-started/troubleshooting/performance/#troubleshooting

**Reply by lunatikk1984** [reply]
> The screenshot looks like you have configured sqlite instead of mariadb. If you are not using our example files you need to do the configuration manually.
> 
> > Still I'd like to know what photoprism is doing in the background.
> 
> I would assume that after importing your files get indexed.
> 
> This troubleshooting guide might also be helpful to solve your locking issues: https://docs.photoprism.app/getting-started/troubleshooting/performance/#troubleshooting

it indexed already during import for the last 6 days. is there any way to see if its really indexing and what the progress is?

best regards,
L

**Reply by graciousgrey** [reply]
It could also be the [background worker](https://docs.photoprism.app/getting-started/config-options/#indexing). You might get more insights by turning on trace logs [and watching the docker logs.](https://docs.photoprism.app/getting-started/troubleshooting/logs/#__tabbed_1_3).
